pact4s/src/test/scala/org/broadinstitute/dsde/workbench/sam/provider/SamProviderSpec.scala

```diff
@@ -0,0 +1,155 @@
+package org.broadinstitute.dsde.workbench.sam.provider
+
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.testkit.ScalatestRouteTest
+import cats.effect.IO
+import cats.effect.unsafe.implicits.global
+import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.workbench.model.{GoogleSubjectId, WorkbenchEmail, WorkbenchUserId}
+import org.broadinstitute.dsde.workbench.oauth2.mock.FakeOpenIDConnectConfiguration
+import org.broadinstitute.dsde.workbench.sam.MockTestSupport.genSamRoutes
+import org.broadinstitute.dsde.workbench.sam.azure.AzureService
+import org.broadinstitute.dsde.workbench.sam.dataAccess.{AccessPolicyDAO, DirectoryDAO}
+import org.broadinstitute.dsde.workbench.sam.google.GoogleExtensions
+import org.broadinstitute.dsde.workbench.sam.model._
+import org.broadinstitute.dsde.workbench.sam.service._
+import org.broadinstitute.dsde.workbench.sam.util.SamRequestContext
+import org.broadinstitute.dsde.workbench.sam.{Generator, MockSamDependencies, MockTestSupport}
+import org.broadinstitute.dsde.workbench.util.health.{StatusCheckResponse, SubsystemStatus, Subsystems}
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito.when
+import org.scalatest.BeforeAndAfterAll
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatestplus.mockito.MockitoSugar.mock
+import pact4s.provider.Authentication.BasicAuth
+import pact4s.provider._
```

Comment:
I'm a Scala novice, so if this is bad advice, feel free to ignore...

I think this imports everything in `pact4s.provider` into the local namespace, right? Would it be better to explicitly import objects by name, so that future readers of this code can more easily see which parts of this spec are Pact-specific?

It won't stop me from approving, just thought it was worth mentioning.

Reply:
In this case, a handful of scala classes are being utilized:

`ProviderInfoBuilder, PactSource, Branch, PublishVerificationResults, ProviderVerificationOption`

They are all from `pact4s.provider._` , it could have been written as

`pact4s.provider.{ProviderInfoBuilder, PactSource, Branch, PublishVerificationResults, ProviderVerificationOption}`

but here we just use the shorthand `pact4s.provider._` 

We will soon have another ticket to refactor boilerplate code, this is when pact-specific code will be lumped together.